### PR TITLE
Ray Integration to Stacks

### PR DESCRIPTION
**Pull request by elamaran11**
This PR adds a new RAY Stack integration to Stacks. This PR includes the following:

    Ray Operator CRDS
    Ray Operator
    Backstage Manifests to Ray Serve ML thru Backstage templates
    ReadMe with instructions

### CODE REVIEW

**Review by punkwalker — CHANGES_REQUESTED** [review]
Thanks @elamaran11 for this stack. I will wait for the requested changes and then approve it.

File: ray-integration/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Ray Integrations
+
+`idpBuilder` is extensible to install Ray Operator for Serving machine learning (ML) models at scale. This integration will help us Ray Kubernetes Operator integrated into our platform which simplifies and accelerates the serving of ML models.
+
+Please use the following command to deploy ray using `idpbuilder`:
+
+```bash
+idpbuilder create \
+  --use-path-routing \
+  --p https://github.com/cnoe-io/stacks//ray-integration
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply ray-integration/README.md]: Fixed.

File: ray-integration/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+Notice that you can add Ray to the reference implementation:
+
+```bash
+idpbuilder create \
+  --use-path-routing \
+  --p https://github.com/cnoe-io/stacks//ref-implementation
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply ray-integration/README.md]: Fixed.

File: ray-integration/backstage-templates/ray-serve/skeleton/manifests/ray-serve.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# Make sure to increase resource requests and limits before using this example in production.
+# For examples with more realistic resource configuration, see
+# ray-cluster.complete.large.yaml and
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply ray-integration/backstage-templates/ray-serve/skeleton/manifests/ray-serve.yaml]: No longer required, references are removed.

File: ray-integration/backstage-templates/ray-serve/sample/pytorch-sample.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        # ===============================
+        ray.train.report(metrics={"loss": test_loss, "accuracy": accuracy})
+
+
+def train_fashion_mnist(num_workers=5, use_gpu=False):
+    global_batch_size = 32
+
+    train_config = {
+        "lr": 1e-3,
+        "epochs": 1, # artifically set low to finish quickly
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply ray-integration/backstage-templates/ray-serve/sample/pytorch-sample.py]: done.

**Comment by elamaran11** [comment]
@punkwalker All required changes and comments are incorporated. Please review and merge.

**Review by punkwalker — APPROVED** [review]
